### PR TITLE
added docs for python api

### PR DESCRIPTION
**Pull request by sidney-tio**
related to #370 task 2

### CODE REVIEW

File: docs/source/getting_started/04_python_mode.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Mastering the Python API
```
Comment (leeping-ng) [inline docs/source/getting_started/04_python_mode.md:1]: Adjust title according to above change

File: docs/source/getting_started/04_python_mode.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+PeekingDuck is designed to be flexible enough for users to deploy PeekingDuck as an app,
+extend it for custom use cases, or just use part of it for prototyping purposes.
+
+This may be particularly useful in the following situations:
```
Comment (leeping-ng) [inline docs/source/getting_started/04_python_mode.md:6]: Here are some examples where...

File: docs/source/getting_started/04_python_mode.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+runner.run()
+
+#Inspect the data
+runner.pipeline.data
+
+```
+
+## Demonstration
+
+We have an in-depth demonstration for running PeekingDuck on Colab [here](https://colab.research.google.com/drive/1gC_qaBSZsyGM1T-L_Vzo_3il44sJBJ2M).
```
Comment (leeping-ng) [inline docs/source/getting_started/04_python_mode.md:45]:
```suggestion
We have an in-depth demonstration for running PeekingDuck on Google Colab [here](https://colab.research.google.com/drive/1gC_qaBSZsyGM1T-L_Vzo_3il44sJBJ2M).
```

File: docs/source/getting_started/04_python_mode.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+1. Wrap a FastAPI app around a Object Detection module
+2. Integrate PeekingDuck into your existing python codebase
+3. Use YOLOv4 for your project in your Jupyter Notebook
+
+
+## Sample Code
+To understand the building blocks, refer to the Introduction [here.](../introduction/01_introduction.md#how-peekingduck-works)
+
+
+You may also refer to our API Reference to understand PeekingDuck's capabilities [here](/peekingduck.pipeline.nodes)
```
Comment (leeping-ng) [inline docs/source/getting_started/04_python_mode.md:17]:
```suggestion
You may also refer to our API Reference to understand PeekingDuck's capabilities [here](/peekingduck.pipeline.nodes).
```

File: README.md [hunk 5]
```diff
@@ -53,6 +53,11 @@ PeekingDuck is an open-source, modular framework in Python, built for Computer V
 
 4. For more help on how to use PeekingDuck's command line interface, you can use `peekingduck --help`.
 
+### Mastering the Python API
```
[the remark this answers is not in the document]
Comment (sidney-tio) [reply README.md:56]: no notebook. we dont want to limit the use case to notebooks; should encourage ppl to build apps around it.

File: README.md [hunk 6]
```diff
@@ -53,6 +53,11 @@ PeekingDuck is an open-source, modular framework in Python, built for Computer V
...
+
+Apart from running it via CLI, PeekingDuck can be ran as a python module as well. See our [guide](docs/source/getting_started/04_python_mode.md) on using PeekingDuck via the Python API.
```
[the remark this answers is not in the document]
Comment (sidney-tio) [reply README.md:58]: sphinx will convert this to a http hyperlink on rtd. should we just maintain this for github?

[the remark this answers is not in the document]
Comment (leeping-ng) [reply README.md:58]: As this the Github readme, we should point to RTD.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (leeping-ng) [reply README.md:56]: That is very true, my bad.

**Pull request merged**